.github/workflows/pytest.yml

```diff
@@ -0,0 +1,42 @@
+name: Test
```

Comment:
Is `Test` a good name for this workflow? Similarly I think the filename should be more representative of it's contents

Reply:
Good spot, I've split this file into two workflows as the single workflow required the conflation of two disparate tasks. Now we have the workflows in their respective files:
* Pytest, ensuring all tests pass.
* Black, ensuring all formatting is correct.